Automattic/pocket-casts-android
# Audio ducking now supported during external notifications

## Description

Added functionality for audio ducking when external notifications come in. There was already some logic there around it hence I've labeled this a fix.

Fixes #529

## Testing Instructions
1. Fresh install of the app.
2. Start the app.
3. Navigate to Profile, then Settings, then Notifications
4. Notice that there is no longer a toggle for 'Play over notifications' and it now launches a dialog
5. Select Never
6. Play a podcast then launch an external audible notification
     - To launch an external audible notification, I used Google maps and asked for turn by turn directions whilst a podcast was playing.
7. Notice that the apps audio stops playing while the audible notification is active and resumes after the notification stops.
8. Select the toggle for 'Play over notifications' again.
9. Select 'Duck volume' and repeat step 6.
10. Notice that the podcast volumes dips to half volume while the external notification plays and resumes full volume after the notification ends.
11. Select the toggle for 'Play over notifications' again.
12. Select 'Always' and repeat step 6.
13. Notice that the podcast audio plays throughout the external notification plays at full volume

## Screenshots or Screencast 
![DuckAudio1](https://github.com/Automattic/pocket-casts-android/assets/1049216/09896f35-ab4f-47fe-9306-6bebea571bc0)
![DuckAudio2](https://github.com/Automattic/pocket-casts-android/assets/1049216/32326253-63f3-4b9a-9554-f79dfb82ee05)
![DuckAudio3](https://github.com/Automattic/pocket-casts-android/assets/1049216/b7650342-47b3-47a2-bff3-e4ace9da1556)

Updated the logs file to show the setting chosen:
![DuckAudio4](https://github.com/Automattic/pocket-casts-android/assets/1049216/148ac9a9-a361-4cc6-aa32-e5b0c3b5af2e)


## Checklist
 
#### I have tested any UI changes...
